db/migrate/20190705091659_delete_unnecessary_proceeding_types.rb

```diff
@@ -0,0 +1,26 @@
+class DeleteUnnecessaryProceedingTypes < ActiveRecord::Migration[5.2]
+  def safe_proceeding_id
+    ProceedingType.find_by(ccms_code: 'DA001').id
+  end
+
+  def proceeding_ids_to_be_deleted
+    ProceedingType.where.not(ccms_code: %w[DA001 DA002 DA003 DA004 DA005 DA006 DA007]).pluck(:id)
+  end
+
+  def up
+    ApplicationProceedingType.where(proceeding_type_id: proceeding_ids_to_be_deleted).update_all(proceeding_type_id: safe_proceeding_id)
+    ProceedingType.delete(proceeding_ids_to_be_deleted)
+    ProceedingType.create(code: 'PR0220',
+                          ccms_code: 'DA020',
+                          ccms_category_law_code: 'MAT',
+                          ccms_category_law: 'Family',
+                          ccms_matter_code: 'MINJN',
+                          ccms_matter: 'Domestic Abuse',
+                          meaning: 'FGM Protection Order-Domestic Abuse ',
+                          description: 'To be represented on an application for a Female Genital Mutilation Protection Order under the Female Genital Mutilation Act.')
+  end
+
+  def down
+    raise ActiveRecord::IrreversibleMigration
+  end
```

Comment:
In my opinion, we should leave this method empty. Which means that you have the option to rollback passed that migration.
Otherwise, with this `ActiveRecord::IrreversibleMigration` you can never rollback more than this migration